lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -96,7 +97,28 @@ def store(credit_card, options={})
         add_stored_credentials(post, credit_card, options)
         add_recurring_contract(post, options)
         add_address(post, options)
-        commit('authorise', post, options)
+
+        initial_response = commit('authorise', post, options)
+	if initial_response.success? == false
+	  initial_response
+        elsif initial_response.authorization.split('#')[2].nil? 
+	  # create failure response
```

Comment:
How do you feel about pulling this out into a helper method? I think it'd be a bit more graceful/easier to read.

Reply:
Done! Thanks for the suggestion!